ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/block/BlockManager.java

```diff
@@ -99,9 +99,65 @@ protected SafeFuture<?> doStop() {
   }
 
   @Override
-  public SafeFuture<BlockImportResult> importBlock(final SignedBeaconBlock block) {
+  public SafeFuture<BlockImportResult> importBlock(
+      final SignedBeaconBlock block, final Optional<BroadcastValidation> broadcastValidation) {
     LOG.trace("Preparing to import block: {}", block::toLogString);
-    return doImportBlock(block, Optional.empty());
+
+    // NO broadcast validation, import the old way
+    if (broadcastValidation.isEmpty()) {
+      return doImportBlock(block, Optional.empty(), Optional.empty());
+    }
+
+    int broadcastValidationLevel = broadcastValidation.get().ordinal();
+
+    // prepare consensus validation future, to me completed by the block import flow
+    final Optional<SafeFuture<BlockImportResult>> consensusValidationResult;
+    if (broadcastValidationLevel > BroadcastValidation.GOSSIP.ordinal()) {
+      consensusValidationResult = Optional.of(new SafeFuture<>());
+    } else {
+      consensusValidationResult = Optional.empty();
+    }
+
+    // GOSSIP
+    SafeFuture<BlockImportResult> importPipeline =
+        validator
```

Comment:
Hmm wondering if it is an idea to make the BlockValidator accept different types of Validation (gossip, consensus, ...) enum as a parameter and then do the logic there. Can be a separate method broadcastValidate() which returns BroadcastValidationResult, can leave the already existing gossip one untouched.

Reply:
yes it is a nice idea, better separation of duties

Reply:
I implemented this spike https://github.com/Consensys/teku/pull/7371/commits/81b0af25a94900c71946237b323e0ba81ff147b0

But I think there is room for improvements following my previous comment. Need also check if my spike is actually correct.